linetools/spectra/xspectrum1d.py

```diff
@@ -517,6 +517,50 @@ def add_noise(self, seed=None, s2n=None, rstate=None):
         #
         return newspec
 
+    def airtovac(self):
+        """ Converts current wavelength array from an assumed air to vacuum wavelength scale
+
+        Returns
+        -------
+        Resultant wavelength array is in AA no matter the input units
+        """
+        # Convert to AA
+        wavelength = self.wavelength.to(u.AA).value
+
+        # Standard conversion format
+        sigma_sq = (1.e4/wavelength)**2. #wavenumber squared
+        factor = 1 + (5.792105e-2/(238.0185-sigma_sq)) + (1.67918e-3/(57.362-sigma_sq))
+        factor = factor*(wavelength>=2000.) + 1.*(wavelength<2000.) #only modify above 2000A
+
+        # Convert
+        wavelength = wavelength*factor
+        # Units
+        new_wave = wavelength*u.AA
+        # Finish
+        self.wavelength = new_wave
+
+    def vactoair(self):
+        """Convert to air-based wavelengths from vacuum
+
+        Returns:
+        ----------
+        Resultant wavelength array is in AA no matter the input units
+        """
+        # Convert to AA
+        wavelength = self.wavelength.to(u.AA).value
+
+        # Standard conversion format
+        sigma_sq = (1.e4/wavelength)**2. #wavenumber squared
+        factor = 1 + (5.792105e-2/(238.0185-sigma_sq)) + (1.67918e-3/(57.362-sigma_sq))
+        factor = factor*(wavelength>=2000.) + 1.*(wavelength<2000.) #only modify above 2000A
+
+        # Convert
+        wavelength = wavelength/factor
+        # Units
+        new_wave = wavelength*u.AA
+        # Finish
+        self.wavelength = new_wave
+
     def constant_sig(self, sigv=0.):
```

Comment:
Finally, it would be nice to have a way to prevent the user from applying these convertions more than once. i.e. imagine that the spectrum is already calibrated at vacuum and airtovac() is applied again that would be problematic. Maybe setting an attribute like self.wvcalib = 'air'/'vacuum' ?